docs/concepts/unions.md

```diff
@@ -76,25 +76,59 @@ print(User(id='456'))  # (2)
 2. We're in lax mode and the numeric string `'123'` is valid as input to the first member of the union, `int`.
    Since that is tried first, we get the surprising result of `id` being an `int` instead of a `str`.
 
+!!! tip
+    For complex cases, it's recommended to use either [`left_to_right`](#left-to-right-mode) mode or [discriminated unions](#discriminated-unions).
```

Comment:
I wonder if it's worth putting a recommendation right in the top section to say something like "always prefer discriminated unions, where possible". And then we can elaborate further on the untagged options and their tradeoffs.